lib/internal/source_map/source_map_cache.js

```diff
@@ -37,8 +37,8 @@ const getModuleSourceMapCache = getLazy(() => {
 // source url magic comments.
 const generatedSourceMapCache = new SafeMap();
 const kLeadingProtocol = /^\w+:\/\//;
-const kSourceMappingURLMagicComment = /\/[*/]#\s+sourceMappingURL=(?<sourceMappingURL>[^\s]+)/g;
-const kSourceURLMagicComment = /\/[*/]#\s+sourceURL=(?<sourceURL>[^\s]+)/g;
+const kSourceMappingURLMagicComment = /(\/[*/]#\s+sourceMappingURL=)(?<sourceMappingURL>[^\s]+)(?![\s\S]*\1)/;
```

Comment:
A minor suggestion: would you mind adding a short comment explaining the regular expressions that it matches the last occurrence?